test/libsolidity/syntaxTests/functionTypes/selector_ternary_contract_public_function.sol

```diff
@@ -0,0 +1,9 @@
+contract C {
+    function f() public pure { }
+    function g() public pure { }
+    function test(bool b) public returns(bytes4) {
+        (b ? C.f : C.g).selector;
+    }
+}
+// ----
+// TypeError 9582: (138-162): Member "selector" not found or not visible after argument-dependent lookup in function () pure.
```

Comment:
@cameel why isn't this working? The test bellow is more or less the same, with the main difference being that here, `f()` and `g()` are declared/defined in the same contract; however, both are public, and thus should have the `.selector` property.

Reply:
In the example below `A` has no relation to `C` (does not inherit from it or anything) so `C.f` is not callable. In the AST it becomes a `Kind::Declaration` and you cannot call it or assign it to a pointer.

On the other hand when these functions are in your contract, they're callable and either `Kind::Internal` or `Kind::External`. In this case it seems that they end up being `Kind::Internal` because they're not explicitly `external` and are not accessed on a contract instance. And you cannot access `.selector` on internal functions. They have no such member.

Reply:
Aaah, OK, that does make sense (although a bit confusing).